### PR TITLE
cli: refactor re-execution into legacy

### PR DESCRIPTION
**Pull request by sergiusens**
This refactor makes it easy to still use the old commands regardless of the
existence of a base in the project and to use the store related commands
from non-legacy regardless of where this is run from.

To achieve this, a new class command inherited from click.Command
was created that can detect if invocation of the legacy code base is
required.

Simple skeletons to handle remote parts, search, update and define,
were created that simply execute into the legacy snapcraft.

Signed-off-by: Sergio Schvezov <sergio.schvezov@canonical.com>

- [ ] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `./runtests.sh static`?
- [ ] Have you successfully run `./runtests.sh tests/unit`?

-----

### CODE REVIEW

**Comment by cjp256** [comment]
The warnings break the unit tests checking output: https://github.com/snapcore/snapcraft/blob/legacy/tests/unit/commands/test_snap.py#L431

File: snapcraft/cli/legacy.py [hunk 1]
```diff
@@ -14,13 +14,39 @@
...
+
+    legacy_python = os.path.join(
+        common.get_legacy_snapcraft_dir(), "usr", "bin", "python3"
+    )
+    legacy_snapcraft = os.path.join(
+        common.get_legacy_snapcraft_dir(), "bin", "snapcraft"
+    )
+
+    cmd = [legacy_python, legacy_snapcraft] + sys.argv[1:]
+    logging.debug("Running legacy snapcraf with: {}".format(cmd))
```
Comment (cjp256) [inline snapcraft/cli/legacy.py]: s/snapcraf/snapcraft

**Comment by cjp256** [comment]
Did some testing: cleanbuild with and without legacy, looked good. Ran unit tests, looked good.

Ran refresh, update, search, and define -- looked good too.

File: snapcraft/cli/__main__.py [hunk 2]
```diff
@@ -42,44 +39,5 @@
...
-        common.get_legacy_snapcraft_dir(), "usr", "bin", "python3"
-    )
-    legacy_snapcraft = os.path.join(
-        common.get_legacy_snapcraft_dir(), "bin", "snapcraft"
-    )
-
-    os.execv(legacy_python, [legacy_python, legacy_snapcraft] + sys.argv[1:])
-
-
-def run():
```
[the remark this answers is not in the document]
Comment (sergiusens) [reply snapcraft/cli/__main__.py]: Yeah, those `import as` can confuse things at time :-)

**Comment by cjp256** [comment]
Did some testing on latest: cleanbuild with and without legacy, refresh, update, search, and define -- all systems seemed fine. LGTM

**Pull request merged**